internal/oidc/idpdiscovery/idp_discovery_handler.go

```diff
@@ -58,6 +59,9 @@ func responseAsJSON(upstreamIDPs oidc.UpstreamIdentityProvidersLister) ([]byte,
 	for _, provider := range upstreamIDPs.GetLDAPIdentityProviders() {
 		r.IDPs = append(r.IDPs, identityProviderResponse{Name: provider.GetName(), Type: idpDiscoveryTypeLDAP})
 	}
+	for _, provider := range upstreamIDPs.GetActiveDirectoryIdentityProviders() {
```

Comment:
This for loop isn't covered by unit test because the unit test only ever adds one AD upstream at a time.

Reply:
Resolved in cc3875f.